### ISSUE TITLE
Error: Ambiguous match found for serviceIdentifier: Symbol(PostMessageService)

### ISSUE THREAD

**Issue by marcin-jaskulski-wttech** [bloomreach/spa-sdk#10]
Hi, 

I'm experimenting with Bloomreach using developers instance. I have React based SPA that uses you SDKs:

```
    "@bloomreach/react-sdk": "18.0.1",
    "@bloomreach/spa-sdk": "18.0.1",
```

I get a lot of these errors:

```Error: Ambiguous match found for serviceIdentifier: Symbol(PostMessageService)```

It's hard to say how to reproduce this issue. It happens from time to time. There are days when it almost never appears and there are days (like today) when this error doesn't let me work efficiently.

My solutions are one of these (or their combination):
* refresh the page
* navigate to a different page and get back
* change frontend url for a channel in project configuration to `Channel front end URL` and get back to localhost:3000
* remove the project and create it again - this is unwelcome since I lose all the introduced changes
Note that these solution don't always help.

Any idea what can be the reason and how to solve it?

![image](https://user-images.githubusercontent.com/103045538/215700374-0b54f425-e454-48f9-bbdc-5977e40be7c7.png)

**Comment by marcin-jaskulski-wttech** [comment]
This is what I can see in a browser logs:
![image](https://user-images.githubusercontent.com/103045538/215704429-fb15fd64-a6a4-4e8d-ad3b-c0cda46c37d0.png)

**Comment by marcin-jaskulski-wttech** [comment]
My hypothesis is that the slower bloomreach responses are, the more often this error appears

**Comment by beetlerom** [comment]
Hello @marcin-jaskulski-wttech thank you for reporting this issue!
@joerideg This sounds like it's related to the async refactoring that we introduced with 18.0.1 . Does this ring a bell?

**Comment by joerideg** [comment]
Created Created https://issues.onehippo.com/browse/SPASDK-162  for progress tracking

**Comment by beetlerom** [comment]
@marcin-jaskulski-wttech Do you have a repository where we can reproduce this issue? We do not experience this error in our tests. Thank you!

**Comment by marcin-jaskulski-wttech** [comment]
Unfortunately, it's not publicly available and I cannot share it. I noticed that the issue happens when the browser developer console is opened. Maybe this will help you reproduce it.

**Comment by jwgmeligmeyling** [comment]
Can confirm this issue, we experience it as well.

**Comment by jwgmeligmeyling** [comment]
In our case it seems to be that our componets get mounted twice and effects run twice as well caused by `<React.StrictMode>` since [React 18.0](https://react.dev/blog/2022/03/29/react-v18).

@marcin-jaskulski-wttech did you recently upgrade to React 18.0 by any chance? Do you use a non-standard setup where the `initialize(configuration)` call is done in a `useEffect(...)` call or the `configuration` is returned from a `useMemo` call?

@beetlerom I think it should be easier to reproduce the issue on React v18.

**Comment by joerideg** [comment]
@mikroware thanks for investigating! That was exactly the problem. I was also able to reproduce it in a unit test by calling initialize twice without waiting for the Promise to be resolved. Since the CmsModule only needs to be initialized once in the lifecycle of the SDK I was able to simply assign the Promise to a variable the first time and await the Promise after and for each subsequent call to the `initialize` method. 

Once we have put my fix through the proper process we will create a release (soon) with the fix and then I will notify you here and close this issue. Thanks for collaborating!

**Comment by joerideg** [comment]
Hi, we just released 21.0.0. This reverts the async behavior and inversify dynamic loading we introduced in 18. 
This should fix your issue 👍🏻
